### DISCUSSION TITLE
Task Template

### DISCUSSION THREAD

**Discussion by evnd04** [uprm-inso4116-2024-2025-s1/semester-project-SafeRUM#4] [Key]
Hello everyone! Find the task template I created for last semester’s project below. While it is not required to use each section you will need to at least provide a title, objective, implementation idea, and a deadline. If these requirements aren't met, we (the managers) will not approve your task, so it is important you follow this structure. If you have any doubts, don't hesitate to comment in this discussion or send me a direct message through any of the modes of communication provided!

**Before adding a description to your issue, please make sure to do the following:**
- Add the assignees that will be working on the issue if you have them in mind already
- Under “Projects”, click/tap SafeRUM and select the iteration that pertains to the relevant milestone. Adding your issue to the SafeRUM _project_ helps managers assign points by the end of the milestone. 
- Add the relevant labels (also referred to as tags internally) for your issue. 

1. Every issue that gets posted needs to be approved by a manager, so please make sure to include the “Pending Approval” tag and the relevant milestone tag before you post your issue.
2. Please make sure to add the team/topic the issue relates to (Documentation, Admin Accounts and Sessions Team, Location Management Team, Trust System Team, User Accounts and Sessions Team).
3. As was done on Milestone 1, add the "Completed" tag when your issue is done and ready to be closed by a manager.
4. All Milestone 1 issues should include the "Milestone 1" label. All Milestone 2 issues should include the "Milestone 2" label. All Milestone 3 issues should include the "Milestone 3" label.

- Under “Milestones”, click/tap the milestone that your issue belongs to.
- Awesome! Now you’re ready to add a description (using the task template below) and add a title! 

===========================

### **Task Template:**

**Objective**:

> Provide an objective for the task (What would the task accomplish if implemented?)

**Task Description**:
	
> A task description is a detailed outline or explanation of a specific job, assignment, or project that needs to be completed. Describe the work that will be done to accomplish the goal of the task.

**Implementation**:

> Provide implementation ideas on how you would implement the feature, this could be added on with time as you work on the feature.

**Subtasks**:

> Preferably split the task but if necessary add a subtask description. Describe the smaller tasks involved in the implementation of your task.

Testing and Debugging:

> How will your feature be tested or debugged? Provide a brief description of that process if necessary.

**Deadline**:

> Set a deadline based on the project timeline and priorities.

===========================

**VERY IMPORTANT NOTE: IF THE TASK IS A LECTURE TOPIC TASK ADD [Lecture Topic Task] IN THE ISSUE TITLE. OTHERWISE, GITHUB WILL _NOT_ RECOGNIZE YOUR ISSUE AS A LECTURE TOPIC TASK.**

**Comment by ghost** [comment]
I think you can make this an actual issue/task template. As in, when devs make a new issue, this template automatically gets filled in.

**Reply by evnd04** [reply]
I’ll verify that later today thank you for the idea!

**Reply by joel-alvarado** [reply]
I believe this require's access to the "Settings" tab in GitHub, which only repo admins can see. Unless the managers have the related permissions, you will probably have to ask the professor either:

- a) Make the managers admin
- b) Add the task template himself

**Reply by evnd04** [reply]
That does seem to be the case based on what I found online I’ll verify permissions later and contact the profesor if these aren’t allowed, if this can’t be done for permission reasons I’ll write it here.

**Reply by evnd04** [reply]
Based on what ive been trying, we managers do not have access to the repo settings to template the issues, i willl comunicate this to the professor at a later date to verify if he can do this for us, for now just copy paste the template into the issues you want to post.
